### ISSUE TITLE
Can't scroll navigation on smaller screen

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#49116]
Can't scroll my sidebar anymore, and access links lower on the screen:

<img width="1440" alt="Screenshot 2021-01-21 at 15 32 19" src="https://user-images.githubusercontent.com/87168/105358092-6046b700-5bfe-11eb-8d76-5053777fd419.png">

If I zoom out, I can see them again:

<img width="1440" alt="Screenshot 2021-01-21 at 15 35 24" src="https://user-images.githubusercontent.com/87168/105358070-5ae96c80-5bfe-11eb-89b3-e3bd0da2f9f1.png">

Before the change:

<img width="1299" alt="Screenshot 2021-01-21 at 19 30 09" src="https://user-images.githubusercontent.com/87168/105388282-24701980-5c1f-11eb-81d1-615b01149170.png">

Site: Atomic
Browser: Firefox.

**Comment by kathrynwp** [comment]
I experienced the same issue in Firefox 84.0.2 on https://wpcourses.com/

I had to manually hack the URL in the address bar reach the full dashboard at https://wpcourses.com/wp-admin

**Comment by sirino** [comment]
I came across a similar issue in the Reader in Google Chrome 87.0.4280:
After interacting with the tags section all the way down, I cannot scroll all the way up in the sidebar any longer. To get back to my following stream, I have to naviaget to https://wordpress.com/read in the address bar.

**Comment by sarayourfriend** [comment]
The same issue happens on mobile, it's nearly impossible to scroll the sidebar in iOS Safari.

**Comment by mrfoxtalbot** [comment]
I noticed this too. If the viewport is shorter than the height of the sidebar menu, there will be no way to access the menu items at the bottom of the sidebar.

![nav-unificattion-vertical-scroll](https://user-images.githubusercontent.com/4452464/107097400-289a5a80-680d-11eb-9cfd-b2e832c08e19.gif)

**Comment by frontdevde** [comment]
The problem with this issue is that so far I can't seem to reliably reproduce it. This makes it hard to figure out the root cause and also test potential fixes. 

Over the course of reloading the page hundreds of times in Firefox for about ~20min, I've been able to reproduce the issue twice. What I did discover on those two occasions was that the `scroll` event listener wasn't added at all, while it was in all other instances. With the scroll event listener not being present, the `handleScroll` function that allows for the sidebar to be scrolled isn't being triggered.

The event listener is only being added when the `config.isEnabled( 'nav-unification')` is true. This currently leads me to the assumption that in the cases where we can't scroll, this check potentially evaluates to false at runtime. I haven't been able to validate this assumption yet though, as the issue hasn't resurfaced since I put debugging in place. 

EDIT: I can confirm that the behaviour I am seeing when intentionally setting `config.isEnabled( 'nav-unification')` to false is consistent with the issue described here.

**Comment by frontdevde** [comment]
OK, seem like the above assumption holds true:
When the issue occurs `config.isEnabled( 'nav-unification')` is evaluating to false at runtime.

I also found a reliable way to reproduce the issue: forcing the initial state hydration to be skipped. When we don't have the initial state hydration, the check always returns false at the time we run it. The check being false results in the scroll event listener not being added which in turn results in the sidebar not scrolling as intended.

![Annotation on 2021-02-10 at 11-33-22](https://user-images.githubusercontent.com/1562646/107498267-d9995000-6b93-11eb-8548-a6041534c8f8.png)

EDIT: I noticed that the `is-nav-unification` class gets added to the DOM despite that check being in the same component. Turns out that it is also `false` the first few times render gets triggered but eventually `true` on subsequent renders. We might want to cosider moving the check from `componentDidMount` to `componentDidUpdate` to ensure the scroll listener gets bound when it should. 

<img width="348" alt="Screenshot 2021-02-10 at 11 49 38" src="https://user-images.githubusercontent.com/1562646/107500480-7c52ce00-6b96-11eb-94d3-4894fdd3c5cc.png">
